tracer/src/Datadog.Trace/TracerManagerFactory.cs

```diff
@@ -264,9 +264,10 @@ protected virtual ITraceSampler GetSampler(ImmutableTracerSettings settings)
                 }
             }
 
-            if (settings.GlobalSamplingRateInternal != null)
+            // ReSharper disable once CompareOfFloatsByEqualityOperator -- We only check if the value was set to its default
+            if ((settings.GlobalSamplingRateInternal ?? 1.0) != 1.0)
             {
-                var globalRate = (float)settings.GlobalSamplingRateInternal;
+                var globalRate = (float)settings.GlobalSamplingRateInternal!.Value;
```

Comment:
If `GlobalSamplingRateInternal` can be made not nullable now that it defaults to `1.0`, that would simplify this code quite a bit.

Reply:
Yeah that's what I did, but then I got a warning because it changes the public API. I doubt anybody will complain, but it's technically a breaking change. I guess we can do it in 3.0.

Reply:
Gotcha, thanks.